akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -146,26 +146,25 @@ trait FileAndResourceDirectives {
   def listDirectoryContents(directories: String*)(implicit renderer: DirectoryRenderer): Route =
     get {
       extractRequestContext { ctx ⇒
-        val path = ctx.unmatchedPath
-        val fullPath = ctx.request.uri.path.toString
-        val matchedLength = fullPath.lastIndexOf(path.toString)
-        require(matchedLength >= 0, s"Unmatched path '$path' wasn't a suffix of full path '$fullPath'. " +
-          "This usually means that ctx.unmatchedPath was manipulated inconsistently " +
-          "with ctx.request.uri.path")
-        val pathPrefix = fullPath.substring(0, matchedLength)
-        val pathString = withTrailingSlash(safeJoinPaths("/", path, ctx.log, '/'))
-        val dirs = directories flatMap { dir ⇒
-          safeDirectoryChildPath(withTrailingSlash(dir), path, ctx.log) match {
```

Comment:
I'm getting chills that we are changing this method again... It LGTM but can someone run the tests on Windows to make sure we tried at least not to introduce another hole here?